worker/worker/status.go

```diff
@@ -105,6 +114,19 @@ func (s *Status) endTranscoding(name string) {
 	statusLock.Unlock()
 }
 
+func (s *Status) endThumbnailGeneration(streamContext *StreamContext) {
+	defer s.SendHeartbeat()
+	statusLock.Lock()
+	s.workload -= costThumbnailGeneration
+	for i := range s.Jobs {
+		if s.Jobs[i] == fmt.Sprintf("transcoding %s", streamContext.getTranscodingFileName()) {
```

Comment:
I am not quite sure but should that be `fmt.Sprintf("generating thumbnail for %s", streamCtx.getTranscodingFileName())`?